### DISCUSSION TITLE
Should I whitelist my system IP addresses?

### DISCUSSION THREAD

**Discussion by superlinuxero** [rspamd/rspamd#3636] [Q&A] [answered]
Hi there.

I have some servers running ISPConfig3 and postfix/amavis. I recently set up a couple of test servers with ISPConfig3 and postfix/Rspamd to know more about it previous to our migration from amavis to Rspamd. Frankly, I'm still struggling to understand how Rspamd works, although I'm beginning to like it a lot.

ISPConfig3 installation creates a /etc/rspamd/local.d/users.conf with, among other things, this entry:

```
whitelist-ip {
priority = 10;
ip = "1.2.3.4";
ip = "1234:1234:1234:1234::";

want_spam = yes;
}
```

I'm not sure, being this a web and mail server, it's wise and convenient to whitelist my both v4 and v6 IPs, sice those IPs are being used by the web server as well. Sending unauthenticated mail is completely disabled to all but essential unix accounts, but even so a PHPMailer script with SMTP authentication could still send, without any effective Rspamd control, unfiltered email messages. Besides, DKIM-signing is bypassed if mail is sent from the web server as you can see in this log entry:

```2021-02-05 21:23:39 #4175(normal) <473566>; task; rspamd_task_write_log: id: <00j4CSFww3d2ARx5cfZgLz1daMZCifMLkel8hXrSuo0@sites.cibernetik.net>, qid: <B2437CF00068>, ip: 1234:1234:1234:1234::, user: testuser@cibernetik.com, from: <testuser@cibernetik.com>, (default: S (no action): [0.00/15.00] []), len: 338, time: 635.887ms, dns req: 0, digest: <73196bcedfe1f5686d1046f5e165970d>, rcpts: <anotheruser@oenus.com>, mime_rcpts: <anotheruser@oenus.com>, settings_id: whitelist-ip```

However, if I disabled the whitelist-ip declaration, messages are DKIM-signed:

```2021-02-05 21:17:39 #3868(normal) <cb5907>; task; rspamd_task_write_log: id: <uXbrwdLL9oWMYZveMav2T72BY5gZzqUdzo8AVHjLHZ0@sites.cibernetik.net>, qid: <A1223CF00068>, ip: 1234:1234:1234:1234::, user: testuser@cibernetik.com, from: <testuser@cibernetik.com>, (default: F (no action): [-0.10/15.00] [MIME_GOOD(-0.10){text/plain;},ARC_NA(0.00){},ASN(0.00){asn:16276, ipnet:2001:41d0::/32, country:FR;},DKIM_SIGNED(0.00){cibernetik.com:s=default;},FROM_EQ_ENVFROM(0.00){},FROM_HAS_DN(0.00){},HAS_PHPMAILER_SIG(0.00){},MIME_TRACE(0.00){0:+;},RCPT_COUNT_ONE(0.00){1;},RCVD_COUNT_ZERO(0.00){0;},TO_DN_NONE(0.00){},TO_MATCH_ENVRCPT_ALL(0.00){}]), len: 338, time: 586.218ms, dns req: 1, digest: <461ffdbea5c9265783f3050df6af014f>, rcpts: <anotheruser@oenus.com>, mime_rcpts: <another@oenus.com>, settings_id: authenticated```

I really need to have all mail DKIM-signed to enable strict DMARC policies.

Is it safe to non-whitelist my IPs in Rspamd? I really don't want them whitelisted...

And just out of curiosity... if I really had to keep those IPs whitelisted, is there any way I could force DKIM-signing emails sent from whitelisted IPs?

Thanks so much for your help, and thanks for such a nice program, too

Ignacio

**Comment by fatalbanana** [comment, marked as answer]
Hi,

> ISPConfig3 installation creates a /etc/rspamd/local.d/users.conf 
> with, among other things, this entry:

It's not a standard config file, but this looks like `settings`: https://rspamd.com/doc/configuration/settings.html

> ```
> whitelist-ip {
> priority = 10;
> ip = "1.2.3.4";
> ip = "1234:1234:1234:1234::";
>
> want_spam = yes;
> }
> ```

`want_spam` is effecting a full bypass of processing.

You could remove this setting and replace it with `symbols_enabled = ["DKIM_SIGNED"]` for DKIM signing only.

> Is it safe to non-whitelist my IPs in Rspamd?

Yes- just be aware that messages could be rejected, or ... according to your configuration.

**Reply by superlinuxero** [reply]
Thanks very much for your answer. I will remove the whitelist tomorrow and check it out
